lib/standalone.js

```diff
@@ -290,6 +300,27 @@ function getFormatterFunction(selected) {
 	return formatters.json;
 }
 
+/**
+ * @param {InternalApi} stylelint
+ * @param {string[]} fileList
+ * @param {string[]} filePaths
+ */
+function filterByExtensions(stylelint, fileList, filePaths) {
+	const { cwd, extensions, globbyOptions } = stylelint._options;
+	const globCWD = globbyOptions?.cwd || cwd;
+	const directories = fileList
+		.filter((p) => fs.existsSync(p) && fs.lstatSync(p).isDirectory())
+		.map((p) => path.relative(globCWD, p));
+	const extList = extensions?.length ? extensions : DEFAULT_EXTENSIONS;
+
+	return filePaths.filter((file) => {
+		if (directories.some((dir) => file.startsWith(dir)))
+			return extList.some((ext) => file.endsWith(`.${ext}`));
```

Comment:
[suggest]

- `fileList` and `filePaths` are confusing to me. How about renaming them? The following is an example.
- How about explicitly passing `extensions` and `cwd` instead of a `stylelint` object?

```suggestion
function filterByExtensions(inputPathList, resolvedFilePaths, extensions, cwd) {
	const directories = inputPathList
		.filter((p) => fs.existsSync(p) && fs.lstatSync(p).isDirectory())
		.map((p) => path.relative(cwd, p));

	return resolvedFilePaths.filter((file) => {
		if (directories.some((dir) => file.startsWith(dir)))
			return extensions.some((ext) => file.endsWith(`.${ext}`));
```

Caller:

```js
filePaths = filterByExtensions(fileList, filePaths, extensions, globCWD);
```

Reply:
> How about explicitly passing `extensions` and `cwd` instead of a `stylelint` object?

I disagree with this, that's why we have the `stylelint` object for. Check the first commit to see why it doesn't scale to pass everything as a separate argument.

> `fileList` and `filePaths` are confusing to me.

I agree.

> How about renaming them? The following is an example.

Ill take it into account.

Reply:
> I disagree with this, that's why we have the `stylelint` object for. Check the first commit to see why it doesn't scale to pass everything as a separate argument.

You may be right outside this module, but I think there is an advantage for a private function inside this module: we can avoid repetitions about `globbyOptions?.cwd` and `DEFAULT_EXTENSIONS`.

Reply:
This part will be refactored so it will be irrelevant. It doesn't work as advertised.
Ill work on it this weekend and fix it.

Reply:
> we can avoid repetitions about `globbyOptions?.cwd` and `DEFAULT_EXTENSIONS`

We cannot avoid reusing `DEFAULT_EXTENSIONS` to cover the `extensions` === `[]` edge case.
Concerning `globbyOptions?.cwd`, I need both `globbyOptions` and `cwd`—in the latest commit.
We would end up with 5 arguments. Just passing `stylelint` is cleaner IMHO.

I think it's more readable with fewer arguments and I try to avoid having too many arguments passed to a function.